api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java

```diff
@@ -490,4 +514,93 @@ public FirecloudNihStatus getNihStatus() {
           }
         });
   }
+
+  @Override
+  public void addProjectToServicePerimeter(String servicePerimeterName, String billingProject) {
+    final String utf8 = StandardCharsets.UTF_8.name();
+
+    // yes this actually gets URL decoded twice
+    // TODO: update Rawls, then FC Orch, then AoU.  See Terra JIRA AS-559
+    final String doublyEncodedName;
+    try {
+      doublyEncodedName = URLEncoder.encode(URLEncoder.encode(servicePerimeterName, utf8), utf8);
+    } catch (UnsupportedEncodingException e) {
+      throw new ServerErrorException(e);
+    }
+
+    ServicePerimetersApi perimetersApi = servicePerimetersApiProvider.get();
+    retryHandler.run(
+        (context) -> {
+          try {
+            perimetersApi.addProjectToServicePerimeter(doublyEncodedName, billingProject);
+          } catch (ApiException e) {
+            String alreadyInMsg =
+                String.format(
+                    "project %s is already in service perimeter %s",
+                    billingProject, servicePerimeterName);
+            if (configProvider.get().featureFlags.badRequestIsOkWhenAlreadyInPerimeter
```

Comment:
According to Terra, it's an error to add a project to a perimeter it's already in.

That makes migrating `bufferBillingProjectsInPerimeter` to false more difficult, since that is what we will try to do for projects which are in the buffer.  Setting `badRequestIsOkWhenAlreadyInPerimeter` to true will quash this error message.  This PR sets it to true in all environments.  We may later wish to turn it off / remove this code.

Reply:
Hmm. Perhaps instead of this, add a billing buffer column to track whether or not the project was created within a perimeter. The "createWorkspace" logic would defer to this column value, rather than looking at the live config. This way the decision is made/persisted transactionally. This avoids complications during a rollback of the feature flag.

Reply:
OK.

After that it's not too much further to record the perimeter in the buffer table ... and then it's another short step to a multi-tier buffer.  We might end up there if the workspace-creation delays from this PR are unacceptable.

Reply:
Yep.. fine if we need to do it for performance (though may be good to check whether this endpoint has been optimized or not). The only longer-term downside here is that Terra is nearly (or already?) finished building their own project buffer, which might mean we could eliminate our own eventually. But if the endpoint is too slow anyways, there may be no way around it.

Reply:
OK - added a boolean field `in_service_perimeter` to the table